docs/architecture.md

```diff
@@ -0,0 +1,4 @@
+# Architecture
```

Comment:
Seems like we can place the architecture content within this md file instead of having an empty page with 2 links